frontend/util.lua

```diff
@@ -400,4 +400,102 @@ function util.splitToArray(str, splitter, capture_empty_entity)
     return result
 end
 
+--- Convert a Unicode codepoint (number) to UTF8 char
+--
+--- @int c Unicode codepoint
+--- @treturn string UTF8 char
+function util.unicodeCodepointToUtf8(c)
+    if c < 128 then
+        return string.char(c)
+    elseif c < 2048 then
+        return string.char(192 + c/64, 128 + c%64)
+    elseif c < 55296 or 57343 < c and c < 65536 then
+        return string.char(224 + c/4096, 128 + c/64%64, 128 + c%64)
+    elseif c < 1114112 then
+        return string.char(240 + c/262144, 128 + c/4096%64, 128 + c/64%64, 128 + c%64)
+    else
+        return util.unicodeCodepointToUtf8(65533) -- U+FFFD REPLACEMENT CHARACTER
+    end
+end
+
+local HTML_ENTITIES_TO_UTF8 = {
+    ["&lt;"] = "<",
+    ["&gt;"] = ">",
+    ["&amp;"] = "&",
+    ["&quot;"] = '"',
+    ["&apos;"] = "'",
+    ["&nbsp;"] = "\xC2\xA0",
+    ["&#(%d+);"] = function(x) return util.unicodeCodepointToUtf8( tonumber(x) ) end,
+    ["&#x(%x+);"] = function(x) return util.unicodeCodepointToUtf8( tonumber(x,16) ) end,
```

Comment:
Well, I tried to roll things up as you suggested, but it got quite confusing, and I couldn't make it as readable as I think it was.
So, I'd like to keep it as it is:
- a simple htmlToPlainText(text) that can be used as is
- a htmlToPlainTextIfHtml(text) that does some cheap detection of HTML according to 2 simple rules :
  - unbalanced < and >
  - encoded HTML tags `"&lt;%a+&gt;"` directly (no need to decode the whole text and check for tags, cheaper to check for encoded tags directly - and i dont want to decode it if there's no tag in it).

To me, it looks more readable, and explicitely describes my flow of thiking.

I added some comments about it being limited and for use with "simple" HTML. When we get wilder input, we may rework it.